src/main/java/com/microsoft/sqlserver/jdbc/IOBuffer.java

```diff
@@ -3684,7 +3684,10 @@ void writeWrappedBytes(byte value[],
         // what remains in the current staging buffer. However, the value must
         // be short enough to fit in an empty buffer.
         assert valueLength <= value.length;
-        assert stagingBuffer.remaining() < valueLength;
+        
+        int remaining = stagingBuffer.remaining();
```

Comment:
In future please try to club commits whenever necessary. Like this one and next one where you are reusing `remaining` variable.